### PR TITLE
Feat/log files

### LINKED ISSUE #388: Move file logging from ceramic-common to package that is not browser side
to package that is not browser side like ceramic-cli

Re: "issues raised trying to build idx-web and self.id because there's an import of fs in ceramic-common (in lib/logger-provider.js), could we remove it please an make sure nothing in ceramic-common is node-specific"

### PR DESCRIPTION
**Pull request by valmack**
Closes #388 

### Motivation

- Move fs library out of ceramic-common (see issue above)

### Changes

- Adds `addPlugin` method to LoggerProvider
- Adds `ceramic-plugins` from which `logToFiles` plugin is passed

### CODE REVIEW

**Comment by valmack** [comment]
@PaulLeCam added you just to ensure the modules being imported/removed from satisfy your requirement

File: packages/ceramic-core/src/ceramic.ts
Comment (valmack) [inline packages/ceramic-core/src/ceramic.ts:137]: @simonovic86 I am having the `init` method return the provider's options here because we add the default options and I think it makes sense to have that passed to the plugin when it gets added.

But this paradigm means there is no single source of truth for the state of options between plugins if we were to add more. So I think it would be best if we can make the options immutable from the perspective of plugins.

File: packages/ceramic-common/src/logger-provider.ts
Comment (PaulLeCam) [inline packages/ceramic-common/src/logger-provider.ts:1]: Could chalk also be moved to the logger plugin please? AFAICT it only targets node.

**Review by PaulLeCam — COMMENTED** [review]
Apart from the `chalk` dependency still in `ceramic-common` LGTM but I'm not familiar with the full flow and logic so better wait for Janko's approval.

File: packages/ceramic-cli/src/ceramic-daemon.ts
Comment (oed) [inline packages/ceramic-cli/src/ceramic-daemon.ts:6]: Naming it to `ceramic-plugins` seems to generic? Isn't it more of a custom logger that is passed as an option to the ceramic instance?

File: packages/ceramic-common/src/logger-provider.ts
Comment (simonovic86) [reply packages/ceramic-common/src/logger-provider.ts:1]: `chalk` dependency is not really necessary. @valmack feel free to remove it

File: packages/ceramic-core/src/ceramic.ts
Comment (simonovic86) [reply packages/ceramic-core/src/ceramic.ts:137]: Yeah, I agree. I think that we can use `Object.freeze()` in the `init` method and prevent options from being changed for now. Makes sense?

**Review by simonovic86 — COMMENTED** [review]
LGTM. I added a couple of comments related to the immutable `opts` and `chalk` dependency.

File: packages/ceramic-cli/src/ceramic-daemon.ts
Comment (valmack) [reply packages/ceramic-cli/src/ceramic-daemon.ts:6]: Yeah I was thinking maybe `ceramic-logger-plugins` is better

**Review by simonovic86 — APPROVED** [review]

**Pull request merged**
